clang/include/clang/AST/Type.h

```diff
@@ -4719,8 +4719,9 @@ class FunctionEffect {
     NonBlocking = 1,
```

Comment:
Something I only thought about just now: is `None` ever actually used? Otherwise, we should probably just remove it and start at 0 so we don’t have to do this `- 1` and `+ 1` dance below, because that seems like a recipe for disaster in terms of off-by-one errors.

(Even if it is used, we should probably just use `std::optional<FunctionEffect::Kind>` instead)

Reply:
(github being confusing again; this is meant to reference the line above that one in case that’s not obvious...)

Reply:
Good idea. I was afraid of a lot of fallout but it was actually quite an easy change to get rid of the sentinel value.